[CSS-Sizing] Fix to respect block size with max-content and min-content in a table

Ahmad-S792 opened the pull request:
#### e180ca76e7e0337d925a231b4d8759369fabda6d
<pre>
[CSS-Sizing] Fix to respect block size with max-content and min-content in a table

<a href="https://bugs.webkit.org/show_bug.cgi?id=261986">https://bugs.webkit.org/show_bug.cgi?id=261986</a>

Reviewed by Antti Koivisto.

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

Partial Merge (only RenderTable.cpp): <a href="https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869">https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869</a>

The specification [1] says to treat the &apos;min-content&apos;, &apos;max-content&apos; and &apos;fit-content&apos; values
as the initial value of the property in question, if said property is a block length property.
In other words, these CSS values really only make a difference in the inline direction.

[1] <a href="https://www.w3.org/TR/css-sizing-3/#column-sizing">https://www.w3.org/TR/css-sizing-3/#column-sizing</a>

Additionally, it syncs latest test copies from WPT upstream:

Upstream Hash: 708e132acaee153316bf9356b0b2ba64c40a2965

* Source/WebCore/rendering/RenderTable.cpp:
(RenderTable::layout):
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1a.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1a-expected.html: Add Test Case Expectation
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1b.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1b-expected.html: Add Test Case Expectation
* LayoutTests/fast/css-intrinsic-dimensions/height-tables-collapsed.html: Removed in favor of WPT test
* LayoutTests/fast/css-intrinsic-dimensions/height-tables-collapsed-expected.html: Removed Expectation
* LayoutTests/fast/css-intrinsic-dimensions/height-css-tables-collapsed.html: Removed in favor of WPT test
* LayoutTests/fast/css-intrinsic-dimensions/height-css-tables-collapsed-expected.html: Removed Expectation

Canonical link: <a href="https://commits.webkit.org/271054@main">https://commits.webkit.org/271054@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/684afea715cef2cbb199921c849e468310fd5ab7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24066 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2178 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25154 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26206 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22171 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24337 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3808 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24549 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22645 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24310 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1712 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20801 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/1606 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1469 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21719 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26795 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21944 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22006 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27964 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1401 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19072 "Passed tests") | [💥 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/28047 "An unexpected error occured. Recent messages:Failed to print configuration; Skipping applying patch since patch_id isn't provided; Checked out pull request") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1422 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/21487 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6553 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1811 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1754 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/resources/import-expectations.json:
```diff
@@ -103,6 +103,8 @@
     "web-platform-tests/css/css-shapes": "import",
     "web-platform-tests/css/css-shapes/test-plan/index.html": "skip",
     "web-platform-tests/css/css-sizing/aspect-ratio": "import",
+    "web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1a.html": "import",
+    "web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1b.html": "import",
```
I'd much prefer if the whole `web-platform-tests/css/css-sizing/` directory was imported in a separate PR. Not just these 2 tests.

Ahmad-S792 committed:
[CSS-Sizing] Fix to respect block size with max-content and min-content in a table

https://bugs.webkit.org/show_bug.cgi?id=261986

Reviewed by Antti Koivisto.

This patch aligns WebKit with Blink / Chromium and Gecko / Firefox.

Partial Merge (only RenderTable.cpp): https://chromium.googlesource.com/chromium/src.git/+/3a0c4adda19bade345ca115fdcfe825acf0b8869

The specification [1] says to treat the 'min-content', 'max-content' and 'fit-content' values
as the initial value of the property in question, if said property is a block length property.
In other words, these CSS values really only make a difference in the inline direction.

[1] https://www.w3.org/TR/css-sizing-3/#column-sizing

Additionally, it syncs latest test copies from WPT upstream:

Upstream Hash: 708e132acaee153316bf9356b0b2ba64c40a2965

* Source/WebCore/rendering/RenderTable.cpp:
(RenderTable::layout):
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1a.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1a-expected.html: Add Test Case Expectation
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1b.html: Add Test Case
* LayoutTests/imported/w3c/web-platform-tests/css/css-sizing/block-size-with-min-or-max-content-table-1b-expected.html: Add Test Case Expectation
* LayoutTests/fast/css-intrinsic-dimensions/height-tables-collapsed.html: Removed in favor of WPT test
* LayoutTests/fast/css-intrinsic-dimensions/height-tables-collapsed-expected.html: Removed Expectation
* LayoutTests/fast/css-intrinsic-dimensions/height-css-tables-collapsed.html: Removed in favor of WPT test
* LayoutTests/fast/css-intrinsic-dimensions/height-css-tables-collapsed-expected.html: Removed Expectation

Canonical link: https://commits.webkit.org/271054@main
